mergify_engine/actions/copy.py

```diff
@@ -121,8 +123,16 @@ def run(self, ctxt: context.Context, rule: rules.EvaluatedRule) -> check_api.Res
             branches.extend(
                 (
                     branch["name"]
-                    for branch in ctxt.client.items(f"{ctxt.base_url}/branches")
-                    if any(map(lambda regex: regex.match(branch["name"]), regexes))
+                    for branch in typing.cast(
+                        typing.List[github_types.GitHubBranch],
+                        ctxt.client.items(f"{ctxt.base_url}/branches"),
+                    )
+                    if any(
+                        map(
+                            lambda regex: regex.match(typing.AnyStr(branch["name"])),
```

Comment:
Did you mean to cast here?

Reply:
I'm a bit confused about why `str` needs to be cast as `AnyStr`, `AnyStr` seems to be `str|bytes`

Reply:
Cast won't work: 
`mergify_engine/actions/copy.py:133: error: Redundant cast to typing.AnyStr?`

The original error I try to fix is: 

`mergify_engine/actions/copy.py:130: error: Argument 1 to "match" of "Pattern" has incompatible type "str"; expected "AnyStr"`

Reply:
Agreed you shouldn't need cast here 🤔

Reply:
Looks like because it can't guess the type of the string in `re.compile`, since it uses generic it can't guess if `T` is `str` or `bytes`.  I fix it by setting type for `re.compile`

Reply:
Funny thing I don't see the error on my env.